sdk/resourcemanager/azure-resourcemanager-containerinstance/src/main/java/com/azure/resourcemanager/containerinstance/models/ContainerGroup.java

```diff
@@ -922,6 +922,78 @@ WithContainerInstanceAttach<ParentT> withReadOnlyVolumeMountSetting(
                     Map<String, String> volumeMountSetting);
             }
 
+            /**
+             * The stage of the container instance definition allowing to specify liveness probe.
+             * Azure Container Instances supports liveness probes so that you can configure your containers within your
+             * container group to restart if critical functionality is not working.
+             *
+             * @param <ParentT> the stage of the parent definition to return to after attaching this definition
+             */
+            interface WithLivenessProbe<ParentT> {
```

Comment:
nit, is it updatable?

Reply:
Tried with serviceClient, and yes it's updatable...

The one complication is that currently we don't have a way to update child container in `ContainerGroup`, i.e. no `ContainerGroup::updateContainerInstance`. Also tried some previous properties like `ports`, and it's also updatable. 

Maybe we do child container update in a separate PR?

Reply:
sure

Reply:
issue logged: https://github.com/Azure/azure-sdk-for-java/issues/32624